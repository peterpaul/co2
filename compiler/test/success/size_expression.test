package compiler.test;

["stdio.h"] int printf(char * format, ...);

class MyObject
{
	int var;
	int fun(int a, int b)
	{
		int c = a + b + var;
		return c;
	}
}

int main(int argc, char[][] argv)
{
	printf("sizeof int: %d\n", sizeof(int));
	printf("sizeof unsigned: %d\n", sizeof(unsigned));
	printf("sizeof float: %d\n", sizeof(float));
	printf("sizeof char: %d\n", sizeof(char));
	printf("sizeof va_list: %d\n", sizeof(va_list));
	printf("sizeof void: %d\n", sizeof(void));
	printf("sizeof void*: %d\n", sizeof(void*));
	printf("sizeof MyObject: %d\n", sizeof(MyObject));
	return 0;
}