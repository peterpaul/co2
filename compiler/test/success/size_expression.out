sizeof int: 4
sizeof unsigned: 4
sizeof float: 4
sizeof char: 1
sizeof va_list: 4
sizeof void: 1
sizeof void*: 4
sizeof MyObject: 4
